cmd/helm/lint_test.go

```diff
@@ -20,35 +20,53 @@ import (
 	"testing"
 )
 
-var (
-	values                       = []byte{}
-	namespace                    = "testNamespace"
-	strict                       = false
-	archivedChartPath            = "testdata/testcharts/compressedchart-0.1.0.tgz"
-	archivedChartPathWithHyphens = "testdata/testcharts/compressedchart-with-hyphens-0.1.0.tgz"
-	invalidArchivedChartPath     = "testdata/testcharts/invalidcompressedchart0.1.0.tgz"
-	chartDirPath                 = "testdata/testcharts/decompressedchart/"
-	chartMissingManifest         = "testdata/testcharts/chart-missing-manifest"
-)
-
 func TestLintChart(t *testing.T) {
-	if _, err := lintChart(chartDirPath, values, namespace, strict); err != nil {
-		t.Errorf("%s", err)
-	}
-
-	if _, err := lintChart(archivedChartPath, values, namespace, strict); err != nil {
-		t.Errorf("%s", err)
+	tests := []struct {
+		name      string
+		chartPath string
+		err       bool
+	}{
+		{
+			name:      "decompressed-chart",
+			chartPath: "testdata/testcharts/decompressedchart/",
+		},
+		{
+			name:      "archived-chart-path",
+			chartPath: "testdata/testcharts/compressedchart-0.1.0.tgz",
+		},
+		{
+			name:      "archived-chart-path-with-hyphens",
+			chartPath: "testdata/testcharts/compressedchart-with-hyphens-0.1.0.tgz",
+		},
+		{
+			name:      "pre-release-chart",
+			chartPath: "testdata/testcharts/pre-release-chart-0.1.0-alpha.tgz",
```

Comment:
I would suggest adding git hashes that can start from zero or ends with zero plus mixed styles like `0.1.0-0196cc0.1` or `0.1.0-1.0196cc0`

Reply:
And also suffixes that look like hexadecimal but they aren't. For example, `0.1.0-0x64000`

Reply:
So because of the new logic, I don't think the name of the tarball matters anymore. Am I misreading your feedback?

Reply:
I was talking only about negative test cases.